### ISSUE TITLE
Better understanding 

### ISSUE THREAD

**Issue by r2d2levrai** [hansungj/TruncatedBackpropagationThroughTime#1]
can you please give your model class, in order to understand better. thanks

**Comment by r2d2levrai** [comment]
for l in range(self.num_layers):
                state_h = states[-1][1][l][0].detach()
                state_c = states[-1][1][l][1].detach() 

what is the shape of states ?  because i got an error regarding the fact that states = ((hx-1,cx-1),(hx,cx)) so the loop is out of range due to it goes through (hx,cx) and not the num_layers
